File: src/test/java/dev/retrotv/crypt/twe/lea/LEACFBTest.java
package dev.retrotv.crypt.twe.lea;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import javax.crypto.spec.IvParameterSpec;
import java.security.Key;

import static org.junit.jupiter.api.Assertions.assertEquals;

class LEACFBTest {

    @Test
    @DisplayName("LEACFB-128 암복호화 테스트")
    void leacfb128_test() throws Exception {
        String message = "The lazy dog jumps over the brown fox!";
        LEACFB lea = new LEACFB128();
        Key key = lea.generateKey();
        IvParameterSpec iv = lea.generateSpec();

        byte[] encryptedData = lea.encrypt(message.getBytes(), key, iv);
        String originalMessage = new String(lea.decrypt(encryptedData, key, iv));

        assertEquals(message, originalMessage);
    }

    @Test
    @DisplayName("LEACFB-192 암복호화 테스트")
    void leacfb192_test() throws Exception {
        String message = "The lazy dog jumps over the brown fox!";
        LEACFB lea = new LEACFB192();
        Key key = lea.generateKey();
        IvParameterSpec iv = lea.generateSpec();

        byte[] encryptedData = lea.encrypt(message.getBytes(), key, iv);
        String originalMessage = new String(lea.decrypt(encryptedData, key, iv));

        assertEquals(message, originalMessage);
    }

    @Test
    @DisplayName("LEACFB-256 암복호화 테스트")
    void leacfb256_test() throws Exception {
        String message = "The lazy dog jumps over the brown fox!";
        LEACFB lea = new LEACFB256();
        Key key = lea.generateKey();
        IvParameterSpec iv = lea.generateSpec();

        byte[] encryptedData = lea.encrypt(message.getBytes(), key, iv);
        String originalMessage = new String(lea.decrypt(encryptedData, key, iv));

        assertEquals(message, originalMessage);
    }
}

File: src/test/kotlin/dev/retrotv/crypto/twe/lea/LEACFBTest.kt
package dev.retrotv.crypto.twe.lea

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

internal class LEACFBTest {
    @Test
    @DisplayName("LEACFB-128 암복호화 테스트")
    fun leacfb128_test() {
        val message = "The lazy dog jumps over the brown fox!"
        val lea = LEACFB(128)
        val key = lea.generateKey()
        val iv = lea.generateSpec()
        val encryptedData = lea.encrypt(message.toByteArray(), key, iv)
        val originalMessage = String(lea.decrypt(encryptedData, key, iv))
        Assertions.assertEquals(message, originalMessage)
    }

    @Test
    @DisplayName("LEACFB-192 암복호화 테스트")
    fun leacfb192_test() {
        val message = "The lazy dog jumps over the brown fox!"
        val lea = LEACFB(192)
        val key = lea.generateKey()
        val iv = lea.generateSpec()
        val encryptedData = lea.encrypt(message.toByteArray(), key, iv)
        val originalMessage = String(lea.decrypt(encryptedData, key, iv))
        Assertions.assertEquals(message, originalMessage)
    }

    @Test
    @DisplayName("LEACFB-256 암복호화 테스트")
    fun leacfb256_test() {
        val message = "The lazy dog jumps over the brown fox!"
        val lea = LEACFB(256)
        val key = lea.generateKey()
        val iv = lea.generateSpec()
        val encryptedData = lea.encrypt(message.toByteArray(), key, iv)
        val originalMessage = String(lea.decrypt(encryptedData, key, iv))
        Assertions.assertEquals(message, originalMessage)
    }
}
